log/packetlogger.go

```diff
@@ -0,0 +1,155 @@
+package log
+
+import (
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithLogLevel sets the log level
+func WithLogLevel(level string) LoggerOption {
+	return func(args *PacketLogger) { args.LogLevel = level }
+}
+
+// WithOutputPaths adds output paths
+func WithOutputPaths(paths []string) LoggerOption {
+	return func(args *PacketLogger) { args.OutputPaths = paths }
+}
+
+// WithServiceName adds a service name a logged field
+func WithServiceName(name string) LoggerOption {
+	return func(args *PacketLogger) { args.ServiceName = name }
+}
+
+// WithKeysAndValues adds extra key/value fields
+func WithKeysAndValues(kvs map[string]interface{}) LoggerOption {
```

Comment:
Also I really like the zap like `keysAndValues []interface{}` instead of maps. It matches the ergonomics of `logr` and zap and how pkg/log already works. We can pay the runtime costs of checking length and mapping to map[string]interface{} since it shouldn't be done very often.

Reply:
copy that, will update